src/coreclr/tools/Common/Compiler/Logging/MessageContainer.cs

```diff
@@ -191,7 +191,7 @@ private static bool IsTrimmableAssembly(ModuleDesc assembly)
                         continue;
 
                     if (!attribute.FixedArguments[0].Type.IsString
-                        || ((string)(attribute.FixedArguments[0].Value)).Equals("IsTrimmable", StringComparison.Ordinal))
+                        || !((string)(attribute.FixedArguments[0].Value)).Equals("IsTrimmable", StringComparison.Ordinal))
```

Comment:
Due to this bug, we have been trimming assemblies with at least one `AssemblyMetadataAttribute` that is not `IsTrimmable` and that has `True` value.

It turns out that a lot of assemblies have random `AssemblyMetadataAttribute` with `True` value.